hvplot/interactive.py

```diff
@@ -512,3 +507,35 @@ def widgets(self):
                 if w not in widgets:
                     widgets.append(w)
         return pn.Column(*widgets)
+
+
+class _hvplot():
+    _converters = tuple(HoloViewsConverter._kind_mapping)
+
+    __slots__ = ["_interactive"]
+
+    def __init__(self, _interactive):
+        self._interactive = _interactive
+
+    def __call__(self, *args, _kind=None, **kwargs):
+        # The underscore in _kind is to be able to overwrite
+        # kind in kwargs, if used with partial. E.g. the following
+        # `dfi.hvplot.scatter('x','y', kind='area')` will return a
+        # scatter plot.
+        new = self._interactive._resolve_accessor()
+        transform = new._transform
+        transform = type(transform)(transform, 'hvplot', accessor=True)
+        if _kind:
+            kwargs["kind"] = _kind
+        dmap = 'kind' not in kwargs or not isinstance(kwargs['kind'], str)
+        return new._clone(transform(*args, **kwargs), dmap=dmap)
+
+    def __getattr__(self, attr):
+        if attr in self._converters:
+            return partial(self, _kind=attr)
+        else:
+            raise AttributeError(f"'hvplot' object has no attribute '{attr}'")
+
+    def __dir__(self):
+        # This function is for autocompletion
+        return self._converters
```

Comment:
Just noting that this list isn't quite correct. For instance when one imports `hvplot.pandas` then the allowed plot kinds, e.g. `df.hvplot.line()` or `df.hvplot.scatter()`, are defined by the `hvPlotTabular` class: https://github.com/holoviz/hvplot/blob/de80fa913b0c18fc044289b7b704993f3a6ac6a3/hvplot/plotting/core.py#L117

Reply:
I have changed this to `self._interactive._obj.hvplot.__all__`, which should be able to separate the kinds used for pandas `DataFrame` and xarray's `Dataset`. 

`self._interactive._obj` is the "original" `DataFrame`/`Dataset`.

Reply:
@philippjfr what do you think of this approach? It will sometimes be inaccurate (i.e. show `.contour` for a DataFrame) and isn't as complete as what you get with `df.hvplot.` as the docstrings aren't patched. But maybe it's anyway better than nothing and good enough as a first step?

Reply:
I'm fine with this for the time being.